src/resmom/mom_main.c

```diff
@@ -1305,7 +1306,7 @@ initialize(void)
 
 		/* search for host */
 		strncpy(rp->key, host, PBS_MAXHOSTNAME);
-		rp->key[PBS_MAXHOSTNAME] = '\0';
+		xxrp.buf[PBS_MAXHOSTNAME + sizeof(AVL_IX_REC)] = '\0';
```

Comment:
Good question. Each entry of the AVL tree has a key. In this case it's the hostname. The default length of the key is defined in avltree.h, but then overridden by the union definition of xxrp. If you look at the definition of struct rectype, you see a comment that the key may be any size. That's why it's allocated last. The rp variable is defined as a pointer to AVL_IX_REC, which is a different size than the union being used. The compiler complains because it thinks you are accessing memory past the end of the structure, when in fact you are referencing the union. We must instead terminate the string by using xxrp directly rather than rp which is pointing to it. I spent some time on this code, and I'm convinced it's correct.

Reply:
Would it be wise to put this explanation in a comment in the code or do you think that isn't really needed?